core-java-modules/core-java-streams-5/src/test/java/com/baeldung/aggregateEx/AggregateExceptionHandlerUnitTest.java

```diff
@@ -0,0 +1,151 @@
+package com.baeldung.aggregateEx;
+
+import com.baeldung.aggregateEx.entity.ExceptionAggregator;
+import com.baeldung.aggregateEx.entity.Result;
+import io.vavr.control.Either;
+import io.vavr.control.Try;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class AggregateExceptionHandlerUnitTest {
+    static Logger logger = LoggerFactory.getLogger(AggregateExceptionHandlerUnitTest.class);
+
+    @Test
+    public void givenExtractedMethod_whenFoundEx_thenSuppressExIntoRuntimeEx() {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        Throwable runEx = Arrays.stream(strings)
+                .map(str -> callProcessThrowsExAndNoOutput(str))
+                .filter(Objects::nonNull)
+                .reduce(new RuntimeException("Errors Occurred"), (o1, o2) -> {
+                    o1.addSuppressed(o2);
+                    return o1;
+                });
+        processExceptions(runEx);
+        assertEquals("Errors Occurred", runEx.getMessage());
+        assertEquals(3, runEx.getSuppressed().length);
+    }
+    @Test
+    public void givenTryCatchInPipeline_whenFoundEx_thenSuppressExIntoRuntimeEx() {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        RuntimeException runEx = Arrays.stream(strings).map(str -> {
+                    try {
+                        processThrowsExAndNoOutput(str);
+                        return null;
+                    } catch (RuntimeException e) {
+                        return e;
+                    }
+                }).filter(Objects::nonNull)
+                .collect(Collectors.collectingAndThen(Collectors.toList(), list -> {
+                    RuntimeException runtimeException = new RuntimeException("Errors Occurred");
+                    list.forEach(runtimeException::addSuppressed);
+                    return runtimeException;
+                }));
+        processExceptions(runEx);
+        assertEquals("Errors Occurred", runEx.getMessage());
+        assertEquals(3, runEx.getSuppressed().length);
+    }
+
+    @Test
+    public void givenProcessMethod_whenStreamResultHasExAndOutput_thenHandleExceptionListAndOutputList() {
+        List<String> strings = List.of("1", "2", "3", "a", "b", "c");
+        Map map = strings.stream()
+                .map(s -> processReturnsExAndOutput(s))
+                .collect(Collectors.partitioningBy(o -> o instanceof RuntimeException, Collectors.toList()));
+        assert(map.containsKey(Boolean.TRUE) && map.containsKey(Boolean.FALSE));
+        handleExceptionsAndOutputs((List<RuntimeException>) map.get(Boolean.TRUE), (List<Integer>)map.get(Boolean.FALSE));
```

Comment:
Handle the case where there were no exceptions or no results using an empty list:

```suggestion
        List<RuntimeException> exceptions = (List<RuntimeException>) map.get(Boolean.TRUE, Collections.emptyList());
        List<Integer> results = (List<Integer>) map.get(Boolean.FALSE, Collections.emptyList());
        handleExceptionsAndOutputs(exceptions, results);
```

Reply:
After changing map declaration to Map<Boolean, List<Object>> map, cannot type caste to List<Integer> or List<RuntimeException>, hence have changed to List<Object>